cradlepoint/pages/4TestPlanDetails.jsx

```diff
@@ -173,3 +173,56 @@ export default function TestPlanDetails(props) {
         </>
     )
 }
+
+export async function getServerSideProps(context) {
+    /* 
+       Gets Data for Test Plan Details
+       TODO: Error Check await call
+       TODO: Refactor out fetch call
+    */
+    const res = await fetch(`${process.env.HOST}/api/getTestPlan?testPlanId=`+context.query.TestPlanId);
+    const testPlanData = await res.json().then((data) => {
+        return {...data[0], };
+    });
+
+    /* 
+       Gets Data for BOM Table
+       TODO: Error Check await call
+       TODO: Refactor out fetch call
+    */
+    var BOM_entries = [];
+    for(var i = 0; i < testPlanData.summaryBOM.length; i++) {
+        const deviceRes = await fetch(`${process.env.HOST}/api/getDevice?deviceId=`
+        +testPlanData.summaryBOM[i].deviceId);
+        const deviceData = await deviceRes.json();
+        const BOM_entry = { 
+            ...deviceData[0],
+            "isOptional": testPlanData.summaryBOM[i].isOptional,
+            "quantity": testPlanData.summaryBOM[i].quantity,
+        };
+        BOM_entries.push(BOM_entry);
+    }
+
+    /* 
+       Gets Data for Test Cases Table
+       TODO: Error Check await call
+       TODO: Refactor out fetch call
+    */
+    const res2 = await fetch(`${process.env.HOST}/api/getTestCasesByTestPlan?testPlanId=`+context.query.TestPlanId);
+    const testCasesData = await res2.json().then((data) => data.map((testCase => {
+        return {
+            "_id": testCase._id,
+            "name": (testCase.name != "")?testCase.name:"N/A",
+            "description": (testCase.name != "")?testCase.name:"N/A",
+            "percentPassed":"__%",
+            "config": (testCase.config != "")?testCase.config:"N/A",
+            // Other Fields not displayed:
+            // "timeEstimate"
+            // "testPlanId"
+            // "BOM"
+        }
+    })));
+    return {
+      props: {...testPlanData, testCasesData, BOM_entries}, // will be passed to the page component as props
```

Comment:
I kinda feel like these should be separate objects so that we won't run into collisions with entries having the same data. We already have testPlanData in an object so I think we should keep it that way for clarity on what each field holds.